### PR TITLE
Add MemoryInformation API to provide more accurate OS free memory statistics

### PR DESCRIPTION
**Pull request by kgibm**
Initial usage by some FAT tests

### CODE REVIEW

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
Last personal build had one error in com.ibm.ws.collective.rest_fat.rw that is unrelated to this change.

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
@OpenLiberty/message-reviewer Please review new messages: https://github.com/OpenLiberty/open-liberty/pull/3576/files#diff-375c9202ea200eb5080a9a063653d5b5

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/OperatingSystem.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                        while ((line = br.readLine()) != null) {
+                            lines.add(line);
+                        }
+                    }
+                }
+            }
+
+            try {
+                int returnCode = process.waitFor();
+                if (returnCode != 0) {
```
[the remark this answers is not in the document]
Comment (gjdeval) [reply dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/OperatingSystem.java]: OK, that makes sense to me.

File: dev/com.ibm.ws.kernel.boot.core/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 2]
```diff
@@ -740,4 +740,20 @@ info.initlogs=Redirecting stdout and stderr to file {0}
...
+memory.information.unavailable.explanation=This function has not been implemented for this operating system.
+memory.information.unavailable.useraction=Use the OperatingSystem API to get the type of operating system and avoid calling this function for this type of operating system.
+
+os.pagesize.unavailable=CWWKE0949E: Page size information is not implemented on this operating system.
+os.pagesize.unavailable.explanation=This function has not been implemented for this operating system.
+os.pagesize.unavailable.useraction=Use the OperatingSystem API to get the type of operating system and avoid calling this function for this type of operating system.
+
+os.execute.error=CWWKE0950E: The execution of {0} failed with return code {1} with output {2}
+os.execute.error.explanation=The program execution is considered to be a failure because of the unexpected return code.
+os.execute.error.useraction=Ensure that the command exists and is on the available PATH.
```
Comment (chirp1) [inline dev/com.ibm.ws.kernel.boot.core/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]: When you say PATH, do you mean the PATH environment variable? If so, how about if you change "Ensure that the command exists and is on the available PATH." to "Ensure that the command exists and that it is on the PATH environment variable.". If you mean a general path, how about "Ensure that the command exists and that it is on an available path.".

File: dev/com.ibm.ws.kernel.boot.core/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops [hunk 3]
```diff
@@ -740,4 +740,20 @@ info.initlogs=Redirecting stdout and stderr to file {0}
...
+memory.information.unavailable=CWWKE0948E: Available memory information is not implemented on this operating system.
+memory.information.unavailable.explanation=This function has not been implemented for this operating system.
+memory.information.unavailable.useraction=Use the OperatingSystem API to get the type of operating system and avoid calling this function for this type of operating system.
+
+os.pagesize.unavailable=CWWKE0949E: Page size information is not implemented on this operating system.
+os.pagesize.unavailable.explanation=This function has not been implemented for this operating system.
+os.pagesize.unavailable.useraction=Use the OperatingSystem API to get the type of operating system and avoid calling this function for this type of operating system.
+
+os.execute.error=CWWKE0950E: The execution of {0} failed with return code {1} with output {2}
+os.execute.error.explanation=The program execution is considered to be a failure because of the unexpected return code.
```
Comment (chirp1) [inline dev/com.ibm.ws.kernel.boot.core/resources/com/ibm/ws/kernel/boot/resources/LauncherMessages.nlsprops]: The wording is sort of a reswizzle of the message text. Could we add some  info that might be useful to the user, i.e. on the types of failures that might occur: "A variety of situations can cause the failure. For example, blah.....".

**Pull request merged**